### ISSUE TITLE
no-unused does not check class members in general?

### ISSUE THREAD

**Issue by Manusan42** [ajafff/tslint-consistent-codestyle#23]
From the documentation page of the "no-unused" rule, it says:

> Does not check private class members.

But considering this snippet:
```ts
export class Test {
  private aaaa = 'c';
  bbbb: string;
  public cccc: string;

  constructor(private dddd, public eeee) { }
}
```
it does not mark any of the members as unused.
Is this intentional or a bug?
If intentional, is this planned as an extension or is there a specific reason to stay with it?

**Comment by ajafff** [comment]
First of all, checking for unused properties only really works for private properties. protected can still be accessed by (yet to come) subclasses and public properties can be accessed from everywhere. It wouldn't make sense to mark them as unused when they are in fact part of the API.

The current implementation does not require (or use) the type checker. That's why you can use it without the `--project` flag. But it comes with the restriction that there is no type information and control flow analysis available. It simply cannot track which property access is a private property (without implementing even more stuff that's already available from the type checker).
There are plans for a future version of the rule to use the type checker if available and fall back to the current behavior if not.

To summarize: it's intentional but not meant to stay that way.

**Comment by Manusan42** [comment]
Thanks for the insights and great work so far.
